mantidproject/mantid
# Grey out masked bins in table view

This PR adds colour highlighting to masked bins in the table view of matrix workspaces.
Also optimises the caching of masked and monitor spectra by holding only the sets of indices, that are masked or monitor, instead of maps of spectra to boolean flags, no matter true or false.

**To test:**


1. Create sample workspace, mask some bins.
2. See how they appear in table view, mouse hover to check the tooltip. 
3. Make some masked bins in a masked or monitor (or both) spectrum to see if the priority ordering makes sense.

Fixes #17820 .



---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
